crates/aptos-crypto/src/webauthn/webauthn_p256_sigs.rs

```diff
@@ -0,0 +1,238 @@
+
+/// A WebAuthn P256 signature >= 165 bytes, <= 200 bytes
+#[derive(DeserializeKey, Clone, SerializeKey)]
+pub struct WebAuthnP256Signature {
+    /// Challenge from [`PublicKeyCredential`](webauthn_rs_core::proto::PublicKeyCredential)
+    /// Should be equal to the SHA3-256 digest of the `RawTransaction` -> 32 bytes
+    pub(crate) actual_challenge: Vec<u8>,
+    /// Binary concatenation of:
+    /// 1. [`authenticator_data_bytes`](webauthn_rs_core::internals::AuthenticatorAssertionResponse) and
+    /// 2. SHA-256 digest of [`client_data_bytes`](webauthn_rs_core::internals::AuthenticatorAssertionResponse)
+    ///
+    /// Note: This is compatible with signatures generated by FIDO U2F
+    /// authenticators. See §6.1.2 FIDO U2F Signature Format Compatibility
+    /// https://www.w3.org/TR/webauthn-2/#sctn-fido-u2f-sig-format-compat
+    ///
+    /// authenticator_data_bytes consists of:
+    /// - rpIdHash -> 32 bytes
+    /// - flags -> 1 byte
+    /// - signCount -> 4 bytes
+    /// - attestedCredentialData -> 0 bytes (not included in authenticatorGetAssertion §6.3.3)
+    /// - extensions -> variable bytes (see [`RequestAuthenticationExtensions`](webauthn_rs_core::proto::RequestAuthenticationExtensions))
+    ///
+    /// Total size -> 37 + 32 >= 69 bytes
+    pub(crate) verification_data: Vec<u8>,
+    /// P256 Signature -> 64 bytes
+    pub(crate) p256_signature: P256Signature,
+}
+
+impl private::Sealed for WebAuthnP256Signature {}
+
+impl WebAuthnP256Signature {
+    /// Serialize a WebAuthnP256Signature.
+    pub fn to_bytes(&self) -> Vec<u8> {
+        serde_json::to_vec(self)
+            .map_err(|_e| CryptoMaterialError::SerializationError)
+            .unwrap()
+    }
+
+    /// Deserialize a WebAuthnP256Signature, without checking for malleability
+    pub(crate) fn from_bytes_unchecked(
+        bytes: &[u8],
+    ) -> Result<WebAuthnP256Signature, CryptoMaterialError> {
+        serde_json::from_slice(bytes).map_err(|_e| CryptoMaterialError::DeserializationError)
+    }
+
+    /// return an all-zero signature (for test only)
+    #[cfg(any(test, feature = "fuzzing"))]
+    pub fn dummy_signature() -> Self {
+        WebAuthnP256Signature {
+            actual_challenge: vec![],
+            verification_data: vec![],
+            p256_signature: P256Signature::from_bytes_unchecked(&[0u8; P256Signature::LENGTH])
+                .unwrap(),
+        }
+    }
+
+    // Is this needed? Not sure how this should be implemented...
+    /// See P256Signature::check_s_malleability
+    pub fn check_s_malleability(bytes: &[u8]) -> Result<(), CryptoMaterialError> {
+        P256Signature::check_s_malleability(bytes)
+    }
+
+    /// If the signature {R,S} does not have S < n/2 where n is the Ristretto255 order, return
+    /// {R,n-S} as the canonical encoding of this signature to prevent malleability attacks. See
+    /// `check_s_malleability` for more detail
+    pub fn make_canonical(&self) -> WebAuthnP256Signature {
```

Comment:
Same as above, this is already handled by our P256 library. We shouldn't need it here